File: 0338-counting-bits/0338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int result[] = new int[n+1];
        for(int i =0;i<=n;i++){
            if(i%2==0) result[i] = result[i/2];
            else result[i] = result[i/2]+1;
        }
        return result;
    }
}

File: 0338-counting-bits/0338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        result = [0]*(n+1)
        for i in range(n+1):
            if i%2 == 0:
                result[i] = result[i//2]
            else:
                result[i] = result[i//2]+1
        return result
        